library/src/blas1/rocblas_copy.hpp

```diff
@@ -46,21 +49,35 @@ rocblas_status rocblas_copy_template(rocblas_handle handle,
                                      rocblas_stride stridey,
                                      rocblas_int    batch_count)
 {
+    RETURN_ZERO_DEVICE_MEMORY_SIZE_IF_QUERIED(handle);
+
     // Quick return if possible.
     if(n <= 0 || !batch_count)
         return rocblas_status_success;
 
+    if(!x || !y)
```

Comment:
We were moving all memory allocation other than quick return to the _impl so the _template is the reusable interface for rocsolver. Otherwise looks good to me.

Reply:
> We were moving all memory allocation other than quick return to the _impl so the _template is the reusable interface for rocsolver. Otherwise looks good to me.

There's no memory allocation in `SCAL`, so the zero queries can be moved to `*_impl`.